docs/secure-communication-agents.asciidoc

```diff
@@ -1,21 +1,352 @@
+
+All other configuration options are described in <<api-key-settings>>.
+
+[[create-api-key]]
+[float]
+=== Create and validate an API key
+
+APM Server provides a command line interface for creating API keys.
+Keys created using this method can only be used for Agent/Server communication.
+The user that creates the API key will need to have the privileges they wish to give to the API key.
+
+While it is also possible to CURL {es} directly to create API keys,
+using this method is not recommended.
+
+[[create-api-key-subcommands]]
+[float]
+==== `apikey` subcommands
+
+include::{libbeat-dir}/command-reference.asciidoc[tag=apikey-subcommands]
+
+[[create-api-key-privileges]]
+[float]
+==== Privileges
+
+There are three unique privileges you can assign to each API keys.
+If privileges are not specified at creation time, the created key will have all privileges. 
+
+* *Agent configuration* - Required for agents to read
+{kibana-ref}/agent-configuration.html[Agent configuration remotely].
+`--agent-config` gives the `config_agent:read` privilege to the created key.
+* *Ingest* - Required for ingesting Agent events.
+`--ingest` gives the `event:write` privilege to the created key.
+* *Sourcemap* - Required for <<sourcemaps,uploading sourcemaps>>.
+`--sourcemap` gives the `sourcemap:write` privilege to the created key.
+
+[[create-api-key-workflow]]
+[float]
+==== API key example workflow
+
+Create an API key with the `create` subcommand.
+
+The following example creates an API key with a `name` of `java-001`,
+and gives the "agent configuration" and "ingest" privileges.
+
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} apikey create --ingest --agent-config --name java-001
+-----
+
+The response will look similar to this:
+
+[source,console-result,subs="attributes,callouts"]
+--------------------------------------------------
+Name ........... java-001
+Expiration ..... never
+Id ............. qT4tz28B1g59zC3uAXfW
+API Key ........ rH55zKd5QT6wvs3UbbkxOA (won't be shown again)
+Credentials .... cVQ0dHoyOEIxZzVDZ3dnMzVWJia3hPQQ==
+--------------------------------------------------
+
+You should always verify the privileges of an API key after creating it.
+Verification can be done using the `verify` subcommand.
+
+The following example verifies that the `java-001` API key has the "agent configuration" and "ingest" privileges.
+
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} apikey verify --agent-config --ingest --credentials cVQ0dHoyOEIxZzVDZ3dnMzVWJia3hPQQ==  
+-----
+
+If the API key has the requested privileges, the response will look similar to this:
+
+[source,console-result,subs="attributes,callouts"]
+--------------------------------------------------
+Authorized for privilege "event:write"...:          Yes
+Authorized for privilege "config_agent:read"...:    Yes
+--------------------------------------------------
+
+To invalidate an API key, use the `invalidate` subcommand.
```

Comment:
maybe mention that this commands might take a bit to have effect due to caching.